js/source/entries/qualitycontrol.js

```diff
@@ -0,0 +1,493 @@
+    if (totalPages > 1) {
+        for (let i = 1; i <= Math.min(totalPages, 3); i++) {
+            const pageButton = document.createElement('button');
+            pageButton.innerHTML = i;
+            pageButton.classList.add('page-button');
+            if (i === currentPage) {
+                pageButton.disabled = true; // Disable the current page button
+            }
+            pageButton.addEventListener('click', () => {
+                currentPage = i;
+                populateOutlierTable(outliersGlobal); // Use the global outliers array
+            });
+            paginationControls.appendChild(pageButton);
+            pageButtons.push(pageButton);
+        }
+
+        // Check if there are more pages and add ellipsis if needed
+        if (totalPages > 3) {
+            const ellipsis = document.createElement('span');
+            ellipsis.innerHTML = '...';
+            paginationControls.appendChild(ellipsis);
+
+            // Last page button
+            const lastPageButton = document.createElement('button');
+            lastPageButton.innerHTML = totalPages;
+            lastPageButton.addEventListener('click', () => {
+                currentPage = totalPages;
+                populateOutlierTable(outliersGlobal); // Use the global outliers array
+            });
+            paginationControls.appendChild(lastPageButton);
+        }
+    }
+}
+
+
+function updateBoxplot() {
+    // Fetch the selected trait and tempfile
+    var trait_selected = $('#trait_select').val();
+    var tempfile = $('#tempfile').html();
+
+    if (!trait_selected || !tempfile || tempfile.trim() === '') {
+        console.log("Either trait or tempfile is missing!");
+        return;
+    }
+
+    const outlierMultiplier = $("#outliers_range").slider("value") || 1.5;
+
+    // Perform an AJAX call to fetch the actual data
+    $.ajax({
+        url: '/ajax/qualitycontrol/grabdata',  // Adjust this URL if needed
+        data: { 'file': tempfile, 'trait': trait_selected },  // Send both tempfile and trait
+        success: function (response) {
+
+            const boxplotData = response.data || [];  // Adjust based on the actual response structure
+            drawBoxplot(boxplotData, trait_selected, outlierMultiplier);
+            const result = drawBoxplot(boxplotData, trait_selected, outlierMultiplier);
+
+            const outliers = result.outliers || [];
+            console.log("Outliers identified:", outliers);
+
+            populateOutlierTable(outliers);
+
+        },
+
+        error: function (jqXHR, textStatus, errorThrown) {
+            console.error('AJAX request failed: ', textStatus, errorThrown);
+            alert('Error fetching data for boxplot: ' + errorThrown);
+        }
+    });
+}
+
+
+
+
+function drawBoxplot(data, selected_trait, outlierMultiplier) {
+    const groupedData = d3.nest()
+        .key(d => d.locationDbId)
+        .entries(data);
+    if (outlierMultiplier === null){outlierMultiplier = 1.5}    
```

Comment:
What happens if the outlierMultiplier is undefined? May be checking for truthiness could be more general???

Reply:
There is a slider bar that changes the outlierMultiplier value, also there is a table showing minimum and maximum for the trait in each location. In this way it will allow users to define the cut off limit.